modules/cluster-logging-forwarding-separate-indices.adoc

```diff
@@ -0,0 +1,69 @@
+// Module is included in the following assemblies:
+//cluster-logging-external
+:_content-type: PROCEDURE
+[id="cluster-logging-forwarding-separate-indices_{context}"]
+= Forwarding JSON logs from containers in the same pod to separate indices
+
+[IMPORTANT]
+====
+JSON formatting of logs varies by application. This feature should be used sparingly to create indices to separate logs with incompatible JSON formats, as creating too many indices will impact performance. To separate logs from different namespaces, or applications with compatible JSON formats, use queries.
+====
+
+You can forward structured logs from different containers within the same pod to different indices. To use this feature, you must configure the pipeline with multi-container support, and annotate the pods.
+
+.Prerequisites
+
+* {logging-title-uc}: 5.5
+
+.Procedure
+. Create or edit a YAML file that defines the `ClusterLogForwarder` CR object:
++
+[source,YAML]
+----
+apiVersion: "logging.openshift.io/v1"
+kind: ClusterLogForwarder
+metadata:
+  name: instance
+  namespace: openshift-logging
+spec:
+  outputDefaults:
+    elasticsearch:
+      enableStructuredContainerLogs: true <1>
+  pipelines:
+  - inputRefs:
+    - application
+    name: application-logs
+    outputRefs:
+    - default <2>
+    parse: json
+----
+<1> Setting the value `enableStructuredContainerLogs` to `true` enables multi-container outputs.
+<2> You can specify alternate outputs or leave the `default` value.
+
+. Create or edit a YAML file that defines the `Pod` CR object:
++
+[source, YAML]
+----
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      annotations:
+        containerType.logging.openshift.io/heavy: heavy <1>
+        containerType.logging.openshift.io/low: low
+    spec:
+      containers:
+      - name: heavy <2>
+        image: heavyimage
+      - name: low
+        image: lowimage
+----
+<1> Example annotation, format: `containerType.logging.openshift.io/container-name: index`
```

Comment:
I don't know the best way to convey this but the collector will write logs to "app-<index>-write".  We generally advise Elasticsearch to be configured such that this is an alias to an indices set up for rollover.  This would mean the logs end up in indices like "app-<index>-00001".  The challenge is if it is not configured this way then it will actually be "app-<index>-write"